tools/src/keygen/Main.hs

```diff
@@ -146,6 +152,37 @@ getAvvmGenesis AvvmStakeOptions {..} = do
 rearrange :: (MonadIO m, MonadFail m, WithLogger m) => FilePath -> m ()
 rearrange msk = mapM_ rearrangeKeyfile =<< liftIO (glob msk)
 
+genPrimaryKey :: (MonadIO m, MonadFail m, WithLogger m) => FilePath -> m ()
+genPrimaryKey path = generateKeyfile True Nothing path >> pure ()
+
+readKey :: (MonadIO m, MonadFail m, WithLogger m) => FilePath -> m ()
+readKey path = do
+    us <- readUserSecret path
+    logInfo $ maybe "No Pimary key"
+                    (("Primary: " <>) . showKeyWithAddressHash) $
+                    view usPrimKey us
+    logInfo $ maybe "No wallet set"
+                    (("Wallet set: " <>) . showKeyWithAddressHash . decryptESK . view wusRootKey) $
+                    view usWalletSet us
+    logInfo $ "Keys: " <> (T.concat $ L.intersperse "\n" $
+                           map (showKeyWithAddressHash . decryptESK) $
+                           view usKeys us)
+    logInfo $ maybe "No vss"
+                    (("Vss PK: " <>) . showPvssKey) $
+                    view usVss us
+
+showKeyWithAddressHash :: SecretKey -> Text
+showKeyWithAddressHash sk = sformat (build%"; address hash: "%build) pk ah
+  where
+    pk = toPublic sk
```

Comment:
In general there's no sense to have `pk` and `ah` here because they are trivial and only used once, but it's minor.